src/components/SlideElementWrapper.js

```diff
@@ -1,41 +1,74 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useSpring, animated } from 'react-spring';
-
-import { SlideContext } from '../hooks/useSlide';
+import { DeckContext } from '../hooks/useDeck';
+import { AnimationMutexContext } from '../hooks/useMutex';
 
 /**
  * SlideElementWrapper provides a component for animating slideElements
  * Anything wrapped inside will be affected by the transition.
  *
  * It is currently using useSpring but ideally we will be able to switch
  * to whatever react-spring hook a user desires!
- * 
+ *
  * Note: Immediate is a React-Spring property that we pass to the animations
  * essentially it skips animations.
  */
 
 const SlideElementWrapper = ({ elementNum, transitionEffect, children }) => {
-  const [state, ,] = React.useContext(SlideContext);
+  const [
+    { reverseDirection, currentSlideElement, immediateElement }
+  ] = React.useContext(DeckContext);
+  const { signal } = React.useContext(AnimationMutexContext);
+  const activeElement = elementNum === currentSlideElement;
+  const upcomingElement =
+    (elementNum > currentSlideElement && !reverseDirection) ||
+    (elementNum < currentSlideElement && reverseDirection);
+  const previousElement =
+    (elementNum < currentSlideElement && !reverseDirection) ||
+    (elementNum > currentSlideElement && reverseDirection);
 
-  const [styleProps, set] = useSpring(() => transitionEffect.from);
+  const [styleProps, set] = useSpring(() => {
+    if ((activeElement || upcomingElement) && !previousElement) {
+      return reverseDirection ? transitionEffect.to : transitionEffect.from;
+    } else if (previousElement) {
+      return reverseDirection ? transitionEffect.from : transitionEffect.to;
+    } else {
+      return transitionEffect.to;
```

Comment:
This else case is a fallback since react spring throws an exception if no transition configuration is hit. But in regular operation this case should not hit, but if it does, just make the element show rather than blow up the app.